### PR TITLE
Adds addresses field to `gettowerinfo`

### PR DESCRIPTION
**Pull request by sr-gi**
This PR adds a new field to the `gettowerinfo` RPC call. The field includes data regarding the interfaces where the public API is being offered (i.e. ipv4 and/or tor).

The rationale for this is having the API endpoint information handy without having to check the logs (this is currently most relevant for the onion address).

Fixes https://github.com/talaia-labs/rust-teos/issues/143

cc/ @jochemin

### CODE REVIEW

**Comment by sr-gi** [comment]
Notice this is a breaking change for the protos, given it modifies them without updating its version. I don't think it is worth a version bump for them at the moment, maybe when more functionality has been added (or after the next major update).

**Comment by sr-gi** [comment]
Something I'd like to discuss here would be how to address proto updates without them being breaking changes. I have really paid much attention to this so far, and maybe we don't want to, but there are ways of making it backward compatible.

https://developers.google.com/protocol-buffers/docs/proto3#updating

**Comment by sr-gi** [comment]
> Notice this is a breaking change for the protos, given it modifies them without updating its version. I don't think it is worth a version bump for them at the moment, maybe when more functionality has been added (or after the next major update).

> Something I'd like to discuss here would be how to address proto updates without them being breaking changes. I haven't really paid much attention to this so far, and maybe we don't want to, but there are ways of making it backward compatible.
>
> https://developers.google.com/protocol-buffers/docs/proto3#updating

I've rebased this and made the change non-breaking. This was breaking due to the field order, and the only good reason for the ordering was having `addresses` close to `tower_id` when serializing in `json` to show the result to the user. I don't think that actually reasonable, if we want a different ordering for the displayed result we can always implement our own serialization.

I've also moved the `AddressType` logic to common so it can be reused by the clients (e.g. to flag a given address as clearnet or tor). This is relevant to address the `Tor` related issues from #102 in a more elegant way.

TL;DR: this is non-breaking now. `AddressType` is now part of `teos-common`.

File: teos/src/api/tor.rs [hunk 1]
```diff
@@ -10,126 +10,154 @@ use torut::control::UnauthenticatedConn;
...
+        service_ready.trigger();
+
+        // NOTE: Needed to keep connection with control port & hidden service running, as soon as we leave
+        // this function the control port stream is dropped and the hidden service is killed
+        loop {
+            sleep(Duration::from_secs(1)).await;
+            if shutdown_signal_tor.is_triggered() {
+                break;
+            }
+        }
```
Comment (mariocynicys) [inline teos/src/api/tor.rs:147]: This can be replace with
```rust
shutdown_signal_tor.await;
```

**Review by mariocynicys — APPROVED** [review]
LGTM :)

File: teos/src/api/tor.rs (hunk 1 above)
Comment (sr-gi) [reply teos/src/api/tor.rs:147]: Indeed, nice one! I never liked that chuck of code

**Comment by sr-gi** [comment]
Should be good to go now

File: teos/src/api/tor.rs [hunk 2]
```diff
@@ -5,131 +5,159 @@ use std::path::PathBuf;
...
+        shutdown_signal_tor.await;
+
+        // // NOTE: Needed to keep connection with control port & hidden service running, as soon as we leave
+        // // this function the control port stream is dropped and the hidden service is killed
+        // loop {
+        //     sleep(Duration::from_secs(1)).await;
+        //     if shutdown_signal_tor.is_triggered() {
+        //         break;
+        //     }
+        // }
```
Comment (mariocynicys) [inline teos/src/api/tor.rs:147]: This is still here :laughing:

**Review by mariocynicys — APPROVED** [review]
LGTM otherwise!

File: teos/src/api/tor.rs (hunk 2 above)
Comment (sr-gi) [reply teos/src/api/tor.rs:147]: Oops
